napoli-server/src/server.rs

```diff
@@ -16,14 +17,51 @@ use crate::model_adapters::{self, get_order_entry_from_add_request};
 
 pub struct NapoliServer {
     db_handle: DatabaseConnection,
+    pub active_order_update_senders: Arc<Mutex<collections::BTreeMap<i32, OrderSender>>>,
 }
 
+type OrderSender = tokio::sync::watch::Sender<tonic::Result<npb::SingleOrderReply>>;
+
 #[tonic::async_trait]
-impl OrderService for NapoliServer {
+impl npb::order_service_server::OrderService for NapoliServer {
+    type StreamOrderUpdatesStream =
+        Pin<Box<dyn Stream<Item = tonic::Result<npb::SingleOrderReply>> + Send>>;
+
+    async fn stream_order_updates(
+        &self,
+        req: tonic::Request<npb::GetOrderRequest>,
+    ) -> tonic::Result<tonic::Response<Self::StreamOrderUpdatesStream>> {
+        println!("stream_order_updates: Got a request: {:?}", req);
+        let order_id = req.into_inner().order_id;
+
+        let initial_order = self
+            .get_order(tonic::Request::new(npb::GetOrderRequest { order_id }))
+            .await?
+            .into_inner();
+
+        let mut sender_mutex = self.active_order_update_senders.lock().await;
+        let sender = sender_mutex.entry(order_id);
+
+        let rx = match sender {
```

Comment:
Being pedantic: `sender_mutex` is actually the value inside the mutex, i.e. the map of update senders.  So i'd rename this slightly:

```suggestion
        let mut senders = self.active_order_update_senders.lock().await;
        let rx = match senders.entry(order_id) {
```